strings/base_coroutine_foundation.h

```diff
@@ -167,7 +167,9 @@ namespace winrt::impl
     private:
         static fire_and_forget cancel_asynchronously(Async async)
         {
+#ifdef WINRT_IMPL_COROUTINES
```

Comment:
Why not call `submit_threadpool_callback` instead of changing the semantics of this function?

Reply:
Yes, I had a note to make this fix better, but this is to unblock "cannot build because of compiler errors" which is significantly more urgent.

Reply:
Could this type just be excluded entirely? Seems only to be used by `operator co_await` further down in the file (which is also conditioned on `WINRT_IMPL_COROUTINES`).

As is, I'd expect this to still give compilation errors, but of the form "function has no return"

Reply:
Okay, I went and did it.

Reply:
Let's try to avoid further feature creep. It's useful to have fixes stand alone in PRs so they can easily be rolled back and reapplied if needed.

Reply:
The previous fix was incomplete. It kept the compiler happy but didn't preserve behavior.

Reply:
@dunhor Ooh, good point. I missed that it's used only by `co_await`. Let me see what happens if I take it out entirely.

Reply:
Got it 👍

Reply:
@dunhor Okay, it works. I had to comment out a boatload of tests that used co_await, but the rest of them build successfully when `WINRT_IMPL_COROUTINES` is disabled.